libs/shared/connected/src/lib/fiskistofa/ShipSearch/ShipSearch.tsx

```diff
@@ -0,0 +1,157 @@
+import { useEffect, useState } from 'react'
+import { useLazyQuery } from '@apollo/client'
+import { useRouter } from 'next/router'
+import {
+  Box,
+  Button,
+  Input,
+  LoadingDots,
+  Table as T,
+  Text,
+} from '@island.is/island-ui/core'
+import { QueryGetShipsArgs, ShipBasicInfo } from '@island.is/web/graphql/schema'
+import { GET_SHIPS_QUERY } from './queries'
+
+interface ShipSearchProps {
+  shipDetailsHref?: string
+  searchStringIsTooShort?: string
+  resultsFound?: string
+}
+
+export const ShipSearch = ({
+  shipDetailsHref = '/s/fiskistofa/skip',
+  searchStringIsTooShort = 'Leitarstrengur þarf að vera a.m.k. 2 stafir',
+  resultsFound = 'Fjöldi skipa:',
+}: ShipSearchProps) => {
+  const [nameInput, setNameInput] = useState('')
+  const [nameInputDuringLastSearch, setNameInputDuringLastSearch] = useState('')
+
+  const [inputError, setInputError] = useState('')
+  const router = useRouter()
+
+  const getShipDetailsHref = (id: number) => {
+    return `${shipDetailsHref}?nr=${id}`
+  }
+
+  const [loadShips, { data, error, loading, called }] = useLazyQuery<
+    { getShips: ShipBasicInfo[] },
+    QueryGetShipsArgs
+  >(GET_SHIPS_QUERY)
+
+  useEffect(() => {
+    if (router?.query?.name) {
+      setNameInput(router.query.name as string)
+      handleShipSearch(router.query.name as string)
+    }
+  }, [router?.query?.name])
+
+  const ships = data?.getShips ?? ([] as ShipBasicInfo[])
+
+  const handleShipSearch = (nameInput: string) => {
+    const nameInputIsNumber = !isNaN(Number(nameInput)) && nameInput.length > 0
+    if (!nameInputIsNumber && nameInput.length < 2) {
+      setInputError(searchStringIsTooShort)
+      return
+    } else {
+      setInputError('')
+    }
+    if (nameInputIsNumber) {
+      router.push(getShipDetailsHref(Number(nameInput)))
+    } else {
+      setNameInputDuringLastSearch(nameInput)
+      loadShips({
+        variables: {
+          input: {
+            shipName: nameInput,
+          },
+        },
+      })
+    }
+  }
+
+  return (
+    <Box>
+      <Input
+        name="skipaskrarnumer-eda-nafn-skips"
+        label="Skipaskrárnúmer eða nafn skips"
+        value={nameInput}
+        onChange={(ev) => setNameInput(ev.target.value)}
+        hasError={inputError.length > 0}
+        errorMessage={inputError}
+        onKeyDown={(ev) => {
+          if (ev.key === 'Enter') {
+            handleShipSearch(nameInput)
+          }
+        }}
+      />
+      <Box marginTop={3}>
+        <Button
+          disabled={
+            nameInput === nameInputDuringLastSearch && nameInput.length > 0
+          }
+          onClick={() => handleShipSearch(nameInput)}
+        >
+          Leita
```

Comment:
i18n missing in this component.

Reply:
I've addressed this issue by having component props (which are directly read from Contentful since we are using the Connected Component slice)